### PR TITLE
samples: mesh: nrf52: Corrected & improved transition implementation

### PR DESCRIPTION
**Pull request by vikrant8052**
1. User can set transition time as Zero & delay as non-zero value.
    In that case, we have to depend upon timer.
2. Added some Kconfig parameters in prj.conf
3. Corrected & improved transition implementation so that
no_transition_work thread will not get triggered twice at the
end of transition.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

### CODE REVIEW

**Comment by vikrant8051** [comment]
In hurry I forgot to test the result for condition where user could set transition time(tt) = 0 & Delay = non-zero value. In such transition, we could not consider it as fully instantaneous so we have to depend upon timer for proper working.

**Comment by vikrant8051** [comment]
Please don't merge as of now.
I've to add one more commit.

**Comment by vikrant8051** [comment]
I've submitted last commit from my side.

**Comment by vikrant8051** [comment]
@trond-snekvik Please review.

File: samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/transition.c [hunk 1]
```diff
@@ -257,44 +267,51 @@ static void light_lightness_actual_work_handler(struct k_work *work)
 {
 	if (ctl->transition->just_started) {
 		ctl->transition->just_started = false;
-		ctl->transition->start_timestamp = k_uptime_get();
+
+		if (ctl->transition->counter == 0U) {
+			update_light_state();
+			k_timer_stop(&ctl->transition->timer);
```
Comment (trond-snekvik) [inline samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/transition.c]: Should this do `ctl->light->current = ctl->light-target` as well?

Comment (trond-snekvik) [reply samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/transition.c]: Right, I see

**Review by trond-snekvik — APPROVED** [review]

**Comment by vikrant8051** [comment]
@jhedberg Please merge.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**